src/test/java/api/requests/RequestsAPICreationTests.java

```diff
@@ -3503,22 +3505,89 @@ void itemCheckOutRecallRequestCreationShouldProduceNotice() {
+    assertEquals(1, borrowerNoticeSize);
+    verifyNumberOfPublishedEvents(NOTICE, 3);
     verifyNumberOfPublishedEvents(NOTICE_ERROR, 0);
-    List<JsonObject> noticeLogContextItemLogs = FakePubSub.getPublishedEventsAsList(byLogEventType(NOTICE));
+
 
     // verify noticeLogContextItemLogs
     validateNoticeLogContextItem(noticeLogContextItemLogs.get(0), item);
     validateNoticeLogContextItem(noticeLogContextItemLogs.get(1), item);
   }
 
+  @Test
+  void itemCheckOutRecallCancelAgainRecallRequestCreationShouldProduceNotice() {
+    configurationsFixture.enableTlrFeature();
+    JsonObject recallToLoaneeConfiguration = new NoticeConfigurationBuilder()
+      .withTemplateId(UUID.randomUUID())
+      .withEventType(NoticeEventType.ITEM_RECALLED.getRepresentation())
+      .create();
+    JsonObject recallRequestToRequesterConfiguration = new NoticeConfigurationBuilder()
+      .withTemplateId(UUID.randomUUID())
+      .withEventType(NoticeEventType.RECALL_REQUEST.getRepresentation())
+      .create();
+
+    NoticePolicyBuilder noticePolicy = new NoticePolicyBuilder()
+      .withName("Policy with recall notice")
+      .withLoanNotices(List.of(recallToLoaneeConfiguration, recallRequestToRequesterConfiguration));
+
+    useFallbackPolicies(
+      loanPoliciesFixture.canCirculateRolling().getId(),
+      requestPoliciesFixture.allowAllRequestPolicy().getId(),
+      noticePoliciesFixture.create(noticePolicy).getId(),
+      overdueFinePoliciesFixture.facultyStandard().getId(),
+      lostItemFeePoliciesFixture.facultyStandard().getId());
+
+    ItemResource item = itemsFixture.basedUponSmallAngryPlanet();
+    IndividualResource requester = usersFixture.steve();
+    IndividualResource borrower = usersFixture.jessica();
+    ZonedDateTime requestDate = ZonedDateTime.of(2017, 7, 22, 10, 22, 54, 0, UTC);
+
+    checkOutFixture.checkOutByBarcode(item, borrower);
+    var recallRequest = requestsFixture.placeItemLevelHoldShelfRequest(item, requester, requestDate, "Recall");
+    requestsFixture.cancelRequest(recallRequest);
+    requestsFixture.placeItemLevelHoldShelfRequest(item, requester, requestDate, "Recall");
+
+    List<JsonObject> noticeLogContextItemLogs = Awaitility.waitAtMost(1, TimeUnit.SECONDS)
+      .until(() -> getPublishedEventsAsList(byLogEventType(NOTICE)), hasSize(4));
+      // notice for the recall is expected
+    verifyNumberOfSentNotices(4);
+    assertEquals(2, getNumberOfRecallNoticesForBorrower(noticeLogContextItemLogs, borrower));
+    verifyNumberOfPublishedEvents(NOTICE, 4);
+    verifyNumberOfPublishedEvents(NOTICE_ERROR, 0);
+
+
+      // verify noticeLogContextItemLogs
+    validateNoticeLogContextItem(noticeLogContextItemLogs.get(0), item);
+    validateNoticeLogContextItem(noticeLogContextItemLogs.get(1), item);
+
+  }
+
+  private Integer getNumberOfRecallNoticesForBorrower(List<JsonObject> noticeLogContextItemLogs, IndividualResource borrower) {
```

Comment:
Few thoughts on this method:
- recall is mentioned in method's name but not in the body
- name suggests that it's counting patron notices, while in fact it is counting notice events (not the same thing)
- you call this method twice and both times use its return value immediately to compare it against expected value, so maybe this method should do the verification too?

```
  private void verifyNumberOfNoticeEventsForUser(UUID userId, int expectedNoticeEventsCount) {
    long noticeEventsCount = getPublishedEventsAsList(byLogEventType(NOTICE))
      .stream()
      .filter(event ->
        new JsonObject(event.getString("eventPayload"))
          .getJsonObject("payload")
          .getString("userId")
          .equals(userId.toString()))
      .count();

    assertThat(noticeEventsCount, is(expectedNoticeEventsCount));
  }

```